### PR TITLE
validate subject parent updates in beforeUpdate hook

### PR DESCRIPTION
**Pull request by jgraff2**

### CODE REVIEW

File: db/dbErrors.js [hunk 1]
```diff
@@ -87,6 +87,16 @@ errors.create({
   explanation: 'If a subject specifies both a parentAbsolutePath and a ' +
     'parentId, the parent specified by both fields need to have the same id.',
 });
+errors.create({
+  scope: exports,
+  code: 10115,
```
Comment (iamigo) [inline db/dbErrors.js]: codes should be unique

File: tests/api/v1/subjects/patchWithParent.js [hunk 2]
```diff
@@ -387,6 +387,39 @@ describe(`tests/api/v1/subjects/patchWithParent.js, PATCH ${path} >`, () => {
...
+        name: NEW_NAME,
+        parentAbsolutePath: _root.name,
+      })
+      .expect(constants.httpStatus.BAD_REQUEST)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+
+        console.log(res.body);
```
Comment (shriramshankar) [inline tests/api/v1/subjects/patchWithParent.js]: Console log here and at L406 in the file. The rest looks good to me.

Comment (jgraff2) [reply tests/api/v1/subjects/patchWithParent.js]: good catch.

**Comment by iamigo** [comment]
@jgraff2 test failures :(

**Comment by jgraff2** [comment]
The tests seem to be flapping a lot since my recent test changes. I will see if I can figure out why.

**Comment by jgraff2** [comment]
This PR should be ok to merge.

**Pull request merged**